src/components/Router/index.js

```diff
@@ -2,19 +2,106 @@ import { h, Component } from 'preact'
 import createHistory from 'history/createBrowserHistory'
 import { bindActionCreators } from 'redux'
 import { connect } from 'react-redux'
+import io from 'socket.io-client'
+
 import { unboundActions, events } from '../../core'
 import {sendScreen} from '../../Tracker'
 import {wrapArray} from '../utils/array'
 import { createComponentList } from './StepComponentMap'
 
 const history = createHistory()
 
-class Router extends Component {
+
+const Router = (props) => {
+  return (
+    props.options.mobileFlow ? <MobileRouter {...props}/> : <DesktopRouter {...props}/>
+  )
+}
+
+class MobileRouter extends Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      token: null,
+      steps: null,
+      socket: io(process.env.DESKTOP_SYNC_URL),
+      roomId: window.location.pathname.substring(1),
+    }
+    this.state.socket.on('config', this.setConfig(props.actions))
+    this.state.socket.emit('join', {room: this.state.roomId})
+    this.requestConfig()
+  }
+
+  requestConfig = () => {
+    this.state.socket.emit('message', {room: this.state.roomId, event: 'get config'})
+  }
+
+  setConfig = (actions) => {
+    return (data) => {
+      const {token, steps, documentType, step} = data
+      this.setState({token, steps, step})
+      actions.setDocumentType(documentType)
+    }
+  }
+
+  render = (props) => {
+    props.options.token = this.state.token
```

Comment:
re writing props is not good practice, they are the parameters of the component, not its state